controllers/feeds/getFeedChat.js

```diff
@@ -212,8 +212,15 @@ const getFeedChatService = async (req, res) => {
     const data = await getstreamService.notificationGetNewFeed(req.userId, req.token);
     const newFeed = [];
 
-    // eslint-disable-next-line no-restricted-syntax
+    let {last_fetch_date = null} = req.query;
+    if (last_fetch_date) {
+      last_fetch_date = new Date(last_fetch_date);
+      last_fetch_date = last_fetch_date.toISOString();
+    }
     for (const feeds of data.results) {
+      if (last_fetch_date && feeds.updated_at < last_fetch_date) {
+        continue;
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The comparison between `feeds.updated_at` and `last_fetch_date` might fail because both are strings and the comparison will be lexicographical, not chronological. To fix this, convert both to Date objects before comparing.

```diff
-    if (last_fetch_date && feeds.updated_at < last_fetch_date) {
+    if (last_fetch_date && new Date(feeds.updated_at) < new Date(last_fetch_date)) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->